src/components/DatePicker.tsx

```diff
@@ -145,47 +183,36 @@ export const DatePicker = (props: Props) => {
               hasFocus={props.hasFocus}
               error={props.error}
               inputType={'date'}
-              value={props.value ? format(props.value, props.dateFormat) : null}
+              value={props.value ? textInputValue : null}
               dataTestId={props.dataTestId}
             />
           ) : (
             <>
               <TextInput
                 hasFocus={props.hasFocus}
                 onChange={() => {}}
+                maxLength={4}
+                pattern={'[1-9][0-9]{3}'}
+                max={'9999-12-31'}
```

Comment:
what does this do?

Reply:
this disables the possibility to ty a year like 123456